### DISCUSSION TITLE
'wrap(...).toPOJO()' not converting members populated through custom types.

### DISCUSSION THREAD

**Discussion by kyllerss** [mikro-orm/mikro-orm#4055] [Q&A]
**Describe the bug**
I have a number of entities which are populated through [custom types](https://mikro-orm.io/docs/custom-types). I am finding that calls to `wrap(entity_obj).toPOJO()` don't convert those members into POJOs. 

Here is the partial result of converting my instance of `TrackEntity` into a POJO:
```
{
  user_account: {
     uuid: '5f928af9-a3d5-461d-ba52-7333277247c2',
     creation_timestamp: 2023-02-16T02:19:15.179Z,
     last_update_timestamp: 2023-02-16T02:19:15.179Z
  },
  ...
  bounds: BoundsData { 
     upper_left: CoordinateData { lat: 42.440502299, lng: -71.34396133 },
     bottom_right: CoordinateData { lat: 42.437000675, lng: -71.3346487 }
  },
  centroid: CoordinateData { lat: 42.4389968, lng: -71.339892134 },
  ...
}
```

In the above example, `user_account` is a member that is of type `UserAccountEntity` and is being converted just fine into a POJO. Members `bounds` and `centroid` are being built through a custom type and are not entities themselves. 

I am more than happy to create a reproduction project if needed.

**Stack trace**
None

**To Reproduce**
1. Fetch an object that has a custom type from the database.
2. Call `wrap(...).toPOJO()`.

**Expected behavior**
Members that are populated by custom types should also be converted into POJOs.

**Additional context**
I found [this other issue report](https://github.com/mikro-orm/mikro-orm/issues/3493), but they don't seem to be related. That said, I could follow a similar approach to creating a custom definition of `toJSON`, but I feel this should be automatically addressed by the framework to ensure others don't run into the same issue. 

Loving the framework! Great job!

**Versions**

| Dependency | Version |
| - | - |
| mikro-orm | 5.6.9 |

**Comment by B4nan** [comment]
You should override the `toJSON` method on your `Type` class, by default it just uses the runtime value:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/types/Type.ts#L55-L57

edit: and [here](https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/serialization/EntitySerializer.ts#L152-L154) is the place where it gets called

**Comment by kyllerss** [comment]
Excellent! Thank you!
